### ISSUE TITLE
Regarding the issue of reproducing accuracy

### ISSUE THREAD

**Issue by PX-Xu** [Natlem/M-HTCN#6]
Dear authors:
We tried to reproduce your excellent work, M-HTCN, in this repository.  However, we failed to produce the results you provided in your paper on the Cityscapes to Foggy_Cityscapes, Rain_Cityscapes scenario. 
Firstly, we constructed the code and datasets by your tutorial, and then we downloaded the weight you provided about cityscapes to foggy_cityscapes, rain_cityscapes scenario. The result is shown following is the same with your paper. The average mAP in the foggy_cityscapes and rain_Cityscapes datasets is 39.2.
![image](https://github.com/user-attachments/assets/67d4f160-6ea8-4188-a495-2c9659487a90)

Then, we use the code you provided to train the model from cityscapes to fg_cityscapes, the hyper-parameter we use is the following:
![image](https://github.com/user-attachments/assets/54b9feef-420e-4faa-9461-54ff8abf0094)

When the training is done. We trained DTM module by exp_train_dtm.py with the following hyper-parameter:
![image](https://github.com/user-attachments/assets/84739189-0a2e-4bd4-8ee5-62382eee7506)

Lastly, we trained  incrementally to rain_cityscapes by exp_traineval_htcn_inc_ida_dtm.py with the following hyper-parameter:
![image](https://github.com/user-attachments/assets/638d4496-5d08-4c3a-a946-08ed0c74631f)

We obtained the final weight (red rectangle):
![image](https://github.com/user-attachments/assets/12ab2f24-a973-4bcc-8433-85679d56717a)

Finally, we exp_eval.py to evaluate the model we try to reproduce, the result is the following:
![024b7dce7fb8afb03bfab24cf7cb4e0](https://github.com/user-attachments/assets/254961c7-c21d-46ed-bbe3-255abea546ab)

The average mAP is 36.2. I wonder which place I did wrong?
I have checked the hyper-parameter and keep they are the same as the paper provided.
Hope to get your reply.

Best wishes.

**Comment by Natlem** [comment]
Hello,

Unfortunately, it would be very difficult for me to help you with this given that it has been more than 3 years for me, I don't remember much nor I have the intermediate training weight anymore. However, I can point you to usual suspects of low performance:
 - If I remember correctly, the original HTCN method already have some problems with variability of accuracy during training which I found that it was very much dependent on GPU/CUDA/Pytorch version... (weird)
 - Verify that the change of learning rate is decreased by a factor of 10 for the last 20k

**Comment by PX-Xu** [comment]
Thank you very much for your prompt reply！
Well. The following picture is my env(rtx3090, cuda11.8 and pytorch 1.8.1)
![image](https://github.com/user-attachments/assets/b1852274-3f38-4f2a-a070-ef0446a77e05)
Indeed, the environment can cause some changes in accuracy. However, if the accuracy of reproducing using the weights you provided is basically consistent with the original paper, I believe that retraining should be able to reproduce the results in the paper.
I checked the last 20k learning rate you mentioned and found that it decreased tenfold. In the original code, at the 5th epoch, the learning rate was multiplied by 0.1, as shown in the following figure. May I ask if the last 20k will only reduce the learning rate in this 5th epoch?
![image](https://github.com/user-attachments/assets/852acc8c-03b3-4069-a4a2-67c3eee20462)
![image](https://github.com/user-attachments/assets/aec77886-9297-4f1b-a3d2-d9701222f705)



Furthermore, I found that using weights from different epochs also has a significant impact on the results. The result of using the weight of the last epoch is as I gave it above: 36.2. I verified the weights of the 5th and 6th rounds and obtained the following results:
The weight average mAP for the 5th round is 38.3：
![image](https://github.com/user-attachments/assets/eb74c9db-913f-4919-bc92-6f476d41ebc6)

The weight average mAP for the 6th round is 36.0：
![image](https://github.com/user-attachments/assets/6f2c7906-5ee5-492f-a311-da67dd9360dc)
Does it seem that there is also an issue of overfitting in training?

Could you please provide some more details about the training?

Hope to get your reply.

Best wishes.

**Comment by PX-Xu** [comment]
Hello, @Natlem :
Can you share the code of your reproduced "Incr. MTDA KD" which is about the paper - "Incremental learning based multi-domain adaptation for object detection"?
